proto/algorithm_params.proto

```diff
@@ -2,11 +2,14 @@ syntax = "proto3";
 
 package bayesmix;
 
+/*
+ * Parameters used in the BaseAlgorithm class and childs.
+ */
 message AlgorithmParams {
-  string algo_id = 1;
-  uint32 rng_seed = 2;
-  uint32 iterations = 3;
-  uint32 burnin = 4;
-  uint32 init_num_clusters = 5;
-  uint32 neal8_n_aux = 6;
+  string algo_id = 1; // Id of the Algorithm. Must match the ones in the AlgorithmId enum
+  uint32 rng_seed = 2; // Seed for the random number generator
+  uint32 iterations = 3; // Total number of iterations of the MCMC chain
+  uint32 burnin = 4; // Number of iterations to discard as burn-in
+  uint32 init_num_clusters = 5; // Number of clusters to initialize the algorithm, only for marginal algorithms
```

Comment:
I don't think it is the case, but maybe its is just out of luck...

in base_algorithm.cc we write

```
mixing->set_num_components(init_num_clusters);
mixing->initialize();
``` 
So it would seem like the number of components is set in the first line. However `BaseMixing::initialize()` is just a wrapper around the derived classes `initialize_state()`. For the TruncatedSbMixing for instance we have that `initialize_state` does 
``` 
num_components = priorcast->num_components();
``` 

Hence when we go back to `BaseAlgorithm::initialize` we have that 
```
unsigned int num_components = mixing->get_num_components();
  for (size_t i = 0; i < num_components - 1; i++) {
    unique_values.push_back(unique_values[0]->clone());
    unique_values[i]->sample_prior();
  }
```
but num_components is not init_num_clusters anymore!

Reply:
On second thoughts, algorithms such as Slice Sampling (that is conditional) might still require to use this value. I updated the documentation accordingly

Reply:
Alright, I forgot that we overwrite `init_num_clusters` in some classes. Still, I agree that it's a good idea to keep it general, and I like your wording